packages/x-date-pickers/src/internals/hooks/usePicker/usePickerValue.ts

```diff
@@ -318,153 +197,148 @@ export const usePickerValue = <
+            });
+
+      const context: PickerChangeHandlerContext<TError> = {
+        validationError,
+      };
+
+      onChange(action.value, context);
+    }
 
-    if (
-      !params.skipOnChangeCall &&
-      !valueManager.areValuesEqual(utils, dateState.committed, params.value)
-    ) {
-      setValue(params.value);
-
-      if (onChange) {
-        const context: PickerChangeHandlerContext<TError> = {
-          validationError:
-            params.contextFromField == null
-              ? validator({
-                  adapter,
-                  value: params.value,
-                  props: { ...props, value: params.value },
-                })
-              : params.contextFromField.validationError,
-        };
-
-        onChange(params.value, context);
-      }
+    if (isCommitted && onAccept) {
+      onAccept(action.value);
     }
 
-    if (params.action === 'acceptAndClose') {
+    if (isClosed) {
       setIsOpen(false);
-      if (
-        onAcceptProp &&
-        !valueManager.areValuesEqual(utils, dateState.resetFallback, params.value)
-      ) {
-        onAcceptProp(params.value);
-      }
     }
   });
 
   React.useEffect(() => {
-    if (isOpen) {
-      // Update all dates in state to equal the current prop value
-      setDate({ action: 'setAll', value, skipOnChangeCall: true });
+    if (isOpen && inValue !== undefined) {
+      setDateState((prev) => ({ ...prev, lastCommittedValue: inValue }));
     }
   }, [isOpen]); // eslint-disable-line react-hooks/exhaustive-deps
 
-  // Set the draft and committed date to equal the new prop value.
-  if (!valueManager.areValuesEqual(utils, dateState.committed, value)) {
-    setDate({ action: 'setCommitted', value, skipOnChangeCall: true });
+  if (
+    inValue !== undefined &&
+    (dateState.lastControlledValue === undefined ||
+      !valueManager.areValuesEqual(utils, dateState.lastControlledValue, inValue))
+  ) {
+    setDateState((prev) => ({ ...prev, lastControlledValue: inValue, draft: inValue }));
   }
 
   const handleClear = useEventCallback(() => {
     // Reset all date in state to the empty value and close picker.
-    setDate({
+    updateDate({
       value: valueManager.emptyValue,
-      action: 'acceptAndClose',
+      name: 'setValueFromAction',
+      pickerAction: 'clear',
     });
   });
 
   const handleAccept = useEventCallback(() => {
     // Set all date in state to equal the current draft value and close picker.
-    setDate({
+    updateDate({
       value: dateState.draft,
-      action: 'acceptAndClose',
+      name: 'setValueFromAction',
+      pickerAction: 'accept',
     });
   });
 
   const handleDismiss = useEventCallback(() => {
     // Set all dates in state to equal the last committed date.
     // e.g. Reset the state to the last committed value.
-    setDate({ value: dateState.committed, action: 'acceptAndClose' });
+    updateDate({ value: dateState.draft, name: 'setValueFromAction', pickerAction: 'dismiss' });
```

Comment:
For me the behavior is still the same
On `master`, the committed date, is the last date returned by the `useControlled` (so the last `props.value` if controlled and the last value passed to `onChange` if not controlled).

___

Behavior with this PR:

If the value has changed since the last publication, we publish (so after that we always have the published value equal to the draft value)

If the value has changed since the last committing, we commit (so after that we always have the committed value equal to the draft value)

=> After his action, the state is full of dates equal to the draft

___

The only diff I see is for shallow calls on the Clock, where indeed on `master` it would not use the draft value if its importance is "shallow".
Not sure it can actually happen, but maybe it would be safer to use `dateState.lastPublishedValue` instead of `dateState.draft` of action bar callbacks.

Reply:
I'm removing the comment because they can't properly describe the behavior now.

Reply:
The difference in behavior mentioned/discovered by Alex is still relevant after the changes. 🤔 
It's probably happening, because in its current state the `DesktopDateTimePicker` never fires `onAccept` ("publishes" the value), because it lacks a time view facilitating it. 🤔 

Hence, technically, this change would be a tiny breaking change in regards to how `cancel` behaves on `DesktopDateTimePicker` (same as clear for the time being).
What do you think about waiting for the digital clock time view before merging this to avoid this potential small issue? 🤔

Reply:
> I was wondering what appends to a partially selected date and it looks like there is a regression with "cancel". If you select a date, in the calendar and then press cancel, the behavior is similar to "clear" which was not the case before.


Are you talking about this comment ?
If so, I gave an answer here https://github.com/mui/mui-x/pull/8312#issuecomment-1495944801, and applied my comment.
It was indeed a behavior change.

___

> in its current state the DesktopDateTimePicker never fires onAccept ("publishes" the value), because it lacks a time view facilitating it.

Yeah, but did it change with this PR ? 
`onAccept` is indeed not very consistent on `DesktopDateTimePicker` due to the lack of views for the time picking.


___

> hence, technically, this change would be a tiny breaking change in regards to how cancel behaves on DesktopDateTimePicker

Not sure to follow which action you are talking about, is it cancel, clear or dismiss ?
But for the three actions, I struggle to see the behavior change you are talking about and the link with the `onAccept` value.

Could you give me a concrete example of a list of actions leading to a different behavior between `master` and this PR ?

Reply:
> Are you talking about this comment ?
> If so, I gave an answer here [#8312 (comment)](https://github.com/mui/mui-x/pull/8312#issuecomment-1495944801), and applied my comment.
> It was indeed a behavior change.

Ah, yes, I missed this comment somehow. 🙈 🤦 

> Yeah, but did it change with this PR ?
> `onAccept` is indeed not very consistent on `DesktopDateTimePicker` due to the lack of views for the time picking.

Nope, it is like that already, I just mentioned `onAccept` as an additional argument, but the root difference/issue is talked about in the previous point.


> Not sure to follow which action you are talking about, is it cancel, clear or dismiss ? But for the three actions, I struggle to see the behavior change you are talking about and the link with the `onAccept` value.
> 
> Could you give me a concrete example of a list of actions leading to a different behavior between `master` and this PR ?

I'm talking about the fact, that on this PR the `cancel` and `clear` actions seem to behave the same.
A concrete example is shown in Alex's video, but in a gist:
- add `cancel` action: `slotProps={{ actionBar: { actions: ['cancel'] } }}`
- open the picker
- select a date
- open the picker again
- click `cancel`

The difference is that on master it does not clear the value, but on this branch it does.

Reply:
Oh my bad, I did not apply the change I proposed in the comment I linked above
It does fix the issue and I think it make more sense.

We can wait for your new UI to be merged if you want :+1: